### ISSUE TITLE
false-positive Unused Import when importing static java function

### ISSUE THREAD

**Issue by Koriit** [pinterest/ktlint#872]
## Expected Behavior
Applying linter rule should not break the compilation.

## Observed Behavior
There is a false-positive unused import. With hard linter check, cannot build because of linter error. If the import is removed cannot build due to compilation error. 

There was no such issue in 0.37.2

Might be related to https://github.com/pinterest/ktlint/issues/845

## Steps to Reproduce
This is tricky as I couldn't reproduce a "clean" scenario. However, this is a sample project that reproduces the issue with a google library that we had in one of my projects and which is affected: https://github.com/Koriit/issues-ktlint-872

## Your Environment

* Version of ktlint used: **0.38.1**
* Name and version (or code for custom task) of integration used (Gradle plugin, Maven plugin, command line, custom Gradle task): **org.jlleitschuh.gradle.ktlint 9.3.0**
* Version of Gradle used (if applicable): **Gradle 6.5.1**
* Operating System and version: **Mac OS X 10.14.6 x86_64**
* Link to your project (if it's a public repository): https://github.com/Koriit/issues-ktlint-872

**Comment by shashachu** [comment]
@Koriit thanks for the report at the sample project. If you need to unblock yourself, the `ktlint-disable` directives should work; see the `README` for details.

**Comment by Koriit** [comment]
Thanks for the suggestion!

**Comment by olegivo** [comment]
I also reprodused the bug when there are static class import and concrete static class method import both:
```
import org.assertj.core.api.Assertions
import org.assertj.core.api.Assertions.assertThat
```
After removing static class import and leaving static method import all works fine.
